library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -533,3 +638,48 @@ impl fmt::Display for ExitStatus {
         }
     }
 }
+
+#[unstable(feature = "linux_pidfd", issue = "none")]
+#[derive(Debug)]
+pub struct PidFd {
```

Comment:
... here `PidFd` is defined in a file that mostly defines OS-specific implementations which are otherwise hidden behind wrapper types defining the public API. And it doesn't have comments as some public API should.

It probably should me moved to a place where the OS-specific extensions are defined.

` process_unix` can just deal with raw `FileDesc` instead.

Reply:
You are right, the `PidFd` type should be in `library/std/src/os/linux/process.rs`. However, I got stuck trying to change that. For the `pidfd(&self) -> &PidFd` method, the public type needs to be stored in the implementation type `Process`. Would that be okay?

Reply:
I'm not sure what exactly the rules on these kinds of things are, but it would be an improvement. If the separation is supposed to be even stronger one could probably just transmute a reference from `FileDesc` as long as the wrapper is `repr(transparent)`.

Reply:
I moved the type in the most recent commit. Are there any guidelines about cyclic dependencies between modules?

Reply:
1 crate = 1 compilation unit, so there shouldn't be a problem with that.